### PR TITLE
Trim the branch prefix from action.GetBranch

### LINKED ISSUE #13979: 404 error when opening a branch on the dashboard
- Gitea version (or commit ref): 1.14.0+dev-381-g633dc80077
- Git version:  2.29.2
- Operating system: ArchLinuxARM (aarch64)
  
  
  
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x ] Yes (provide example URL) https://try.gitea.io/atomlong/WiFiManager/src/branch/refs/heads/master
  - [ ] No
- Log gist:

## Description

1) Push master branch to https://try.gitea.io/atomlong/WiFiManager
2) open the dashboard on page https://try.gitea.io
3) click "refs/heads/master" link
image
4) "Page Not Found" will come
image

My guess is that the branch name on the Dashboard panel is wrong, it should be 'master', not 'refs/heads/master'.

### PR DESCRIPTION
**Pull request by zeripath**
#13882 has revealed that the refname of an action is actually only a
refname pattern and necessarily a branch. For examplem pushing to
refs/heads/master will result in action with refname refs/heads/master
but pushing to master will result in a refname master.

The simplest solution to providing a fix here is to trim the prefix
therefore this PR proposes this.

Fix #13979 

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by a1012112796** [comment]
@zeripath Please fix lint :)

File: models/action.go [hunk 1]
```diff
@@ -238,7 +239,7 @@ func (a *Action) getCommentLink(e Engine) string {
 
 // GetBranch returns the action's repository branch.
 func (a *Action) GetBranch() string {
-	return a.RefName
+	return strings.TrimPrefix(git.BranchPrefix, a.RefName)
```
Comment (a1012112796) [inline models/action.go]:
```suggestion
	return strings.TrimPrefix(a.RefName, git.BranchPrefix)
```

Comment (zeripath) [reply models/action.go]: that's what I get for doing things quickly without checking them!

**Review by mrsdizzie — APPROVED** [review]

**Review by a1012112796 — APPROVED** [review]

**Comment by lafriks** [comment]
Please send backport

**Pull request merged**